baseband/mark4/tests/test_mark4.py

```diff
@@ -158,18 +168,43 @@ def test_header(self, tmpdir):
         assert header10 == header9
         header11 = Mark4TrackHeader.fromvalues(decade=2010, **header9)
         assert header11 == header9
-        with pytest.raises(AssertionError):  # missing decade
-            Mark4TrackHeader.fromvalues(**header9)
+        # Check passing decade=None still reads the header, and we can set
+        # decade afterward.
+        with open(SAMPLE_FILE, 'rb') as fh:
+            fh.seek(0xa88)
+            header12 = mark4.Mark4Header.fromfile(fh, ntrack=64)
+            assert header12.decade is None
+            header12.decade = 2010
+            assert header12 == header
         with pytest.raises(IndexError):
             header[65]
         with pytest.raises(ValueError):
             header[np.array([[0, 1], [2, 3]])]
         # check that one can construct crazy headers, even if not much works.
-        header12 = mark4.Mark4Header(None, ntrack=53, decade=2010,
+        header13 = mark4.Mark4Header(None, ntrack=53, decade=2010,
                                      verify=False)
-        header12.time = header.time
-        assert header12.ntrack == 53
-        assert abs(header12.time - header.time) < 1. * u.ns
+        header13.time = header.time
+        assert header13.ntrack == 53
+        assert abs(header13.time - header.time) < 1. * u.ns
+
+    def test_infer_decade(self):
+        # Check that infer_decade returns proper decade for
+        # ref_time.year - 5 < year <= ref_time.year + 5.
+        decade = mark4.header.Mark4Header.infer_decade(
```

Comment:
With my suggestion to make this a method that overwrites `decade`, these test would need to be rewritten. But they can just start with `header = Mark4Header(ntrack=16)` or so.